account_reconcile_model_multicompany_propagate/models/account_reconcile_model.py

```diff
@@ -0,0 +1,183 @@
+            "match_text_location_label",
+            "match_text_location_note",
+            "match_text_location_reference",
+            "match_nature",
+            "match_amount",
+            "match_amount_min",
+            "match_amount_max",
+            "match_label",
+            "match_label_param",
+            "match_note",
+            "match_note_param",
+            "match_transaction_type",
+            "match_transaction_type_param",
+            "match_same_currency",
+            "allow_payment_tolerance",
+            "payment_tolerance_param",
+            "payment_tolerance_type",
+            "match_partner",
+            "match_partner_ids",
+            "match_partner_category_ids",
+            "past_months_limit",
+            "number_entries",
+        ]
+
+    def propagate_multicompany_account_reconcile_model(self):
+        """Set the same value for all companies with same chart template."""
+        alien_companies = self.env.user.company_ids - self.env.company
+        if not alien_companies:
+            raise UserError(
+                _(
+                    "There are no other companies to propagate to. "
+                    "Make sure you have access to other companies."
+                )
+            )
+        # Gain access to all companies of current user
+        self = self.with_context(
+            allowed_company_ids=self.env.company.ids + alien_companies.ids
+        )
+
+        companies_chart_template_equal = alien_companies.filtered(
+            lambda c: c.chart_template_id == self.env.company.chart_template_id
+        )
+
+        # Map accounts by company and code
+        accounts = self.env["account.account"].search(
+            [
+                ("company_id", "in", companies_chart_template_equal.ids),
+                (
+                    "code",
+                    "in",
+                    self.line_ids.account_id.mapped("code"),
+                ),
+            ]
+        )
+        accounts_map = defaultdict(dict)
+        for account in accounts:
+            accounts_map[account.company_id.id][account.code] = account.id
+
+        # Map taxes by company and name
+        taxes = self.env["account.tax"].search(
+            [
+                ("company_id", "in", companies_chart_template_equal.ids),
+                (
+                    "name",
+                    "in",
+                    self.line_ids.tax_ids.mapped("name"),
+                ),
+            ]
+        )
+        taxes_map = defaultdict(dict)
+        for tax in taxes:
+            taxes_map[account.company_id.id][tax.name] = tax.id
+
+        # Propagate to other companies
+        for company in companies_chart_template_equal:
+            rec_model = self.search(
+                [("company_id", "=", company.id), ("name", "=", self.name)]
+            )
+            if rec_model:
+                for field in self._propagated_fields():
+                    rec_model.write({field: self[field]})
```

Comment:
This is the same, but a bit more readable:

```suggestion
                    rec_model[field] = self[field]
```

Reply:
It's true. Changed.